### ISSUE TITLE
linter: Mismatched line numbers for `<script>` content

### ISSUE THREAD

**Issue by cordis-dev** [oxc-project/oxc#9333]
### What version of Oxlint are you using?

0.15.12

### What command did you run?

`oxlint-win32-x64.exe t.vue`

### What does your `.oxlintrc.json` config file look like?

N/A

### What happened?

Given this `t.vue` file:

```vue
<template>
    <h1>Hello, Vue!</h1>
    <slot />
</template>

<script>
defineProps({
    title: {
        type: String,
        default: "Test Page"
    }
})
debugger;
</script>
```

Oxlint reports wrong line number:
```
  ⚠ eslint(no-debugger): `debugger` statement is not allowed
   ╭─[t.vue:8:1]
 7 │ })
 8 │ debugger;
   · ─────────
   ╰────
```

It should report this issue for line `13` instead. Now it treats `<script>` tag as the starting point.

Same issue applies for `<script>` tags in astro and svelte files.

**Comment by Sysix** [comment]
@overlookmotel & @Boshen I looked into the problem. 
The `*PartialLoader` for `vue`, `svelte` and `astro` have the start and end index for the script tag.
The source code is passed into the ast-parser. There it will begin the index from zero.
Can you help me to solve this issue or overtake it :)? 
My first try failed^^"
